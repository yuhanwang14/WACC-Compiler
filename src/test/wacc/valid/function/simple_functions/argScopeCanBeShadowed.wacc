# Arguments can be shadowed by the function body

# Output:
# true
#

#Program:

begin
  bool f(int x) is
    bool x = true ;
    return x
  end

  bool x = call f(5);
  println x

end
