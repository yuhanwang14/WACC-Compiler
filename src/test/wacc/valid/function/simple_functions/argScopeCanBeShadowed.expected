true


Program:

begin
  bool f(int x) is
    bool x = true ;
    return x
  end

  bool x = call f(5);
  println x

end